### PR TITLE
Improve migration error message with uninstall command

### PR DESCRIPTION
**Pull request by digitaltom**
## Description

Improve migration error message with uninstall command, 
only raise error when extensions have no successor. 
A base without successor is expected, ie. latest product like SLES 15 SP3.

Fixes https://trello.com/c/r6zomyoz/1652-raise-error-message-in-migration-when-there-is-no-successor

## Change Type

*Please select the correct option.*

- [x **Bug Fix** (a non-breaking change which fixes an issue)
- [ ] **New Feature** (a non-breaking change which adds new functionality)
- [ ] **Documentation Update** (a change which only updates documentation)

## Checklist

*Please check off each item if the requirement is met.*

- [ ] I have verified that my code follows RMT's coding standards with `rubocop`.
- [ ] I have reviewed my own code and believe that it's ready for an external review.
- [ ] I have provided comments for any hard-to-understand code.
- [ ] I have documented the `MANUAL.md` file with any changes to the user experience.
- [ ] RMT's test coverage remains at 100%.
- [ ] If my changes are non-trivial, I have added a changelog entry to notify users at `package/obs/rmt-server.changes`.

## Other Notes

Please use this space to provide notes or thoughts to the team, such as tips on how to review/demo your changes.

### CODE REVIEW

File: spec/models/migration_engine_spec.rb [hunk 1]
```diff
@@ -280,7 +280,7 @@
         it 'raises error with help message' do
           expect { migrations }.to raise_error do |error|
             expect(error).to be_a(MigrationEngine::MigrationEngineError)
-            expect(error.data).to eq(sle12ltss.friendly_name)
+            expect(error.data).to include(sle12ltss.friendly_name)
```
Comment (felixsch) [inline spec/models/migration_engine_spec.rb:283]: I would add additionally to check the context as well but this is nitpicking here:
```suggestion
            expect(error.message).to match(/There are activated extensions\/modules on this system which cannot be migrated/)
            expect(error.data).to include(sle12ltss.friendly_name)
```

**Review by felixsch — APPROVED** [review]
lgtm :+1: 

**How I reviewed this pull request:**

- Run a SLE12LTSS installation against the branch (provided by @digitaltom  on our reference host) and checked the results
- Run spec on my local machine and checked the output
- Looked into the tests if all cases are reflected

** As always, if you think I'm missing something I should tests or review, please let me know! **

**Comment by suse-tests-pass** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Unless this is a trivial change, please include a CHANGELOG entry.<br />
Run <code>osc vc</code> in the <code>package</code> directory to add one.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/187334/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged fe50ad202755ae613bb967feb102a60702b29463 into 05412b52a3cbaf2f3886fe77f84685d3f71aee40)

**Pull request merged**